markup.go

```diff
@@ -170,7 +170,19 @@ func If(cond bool, markup ...MarkupOrComponentOrHTML) MarkupOrComponentOrHTML {
 	if cond {
 		return List(markup)
 	}
-	return nil
+	var result []MarkupOrComponentOrHTML
+	for _, m := range markup {
+		if m == nil {
+			continue
+		}
+		switch m.(type) {
+		case *HTML, Component:
+			result = append(result, Tag("noscript"))
+		default:
+			return nil
```

Comment:
I'm not sure why the default here is `return nil`, because I think this would lead to invalid behavior.

The argument type to `If` is `...MarkupOrComponentOrHTML`, and you only handle `*HTML, Component` explicitly above. That means this `default` case would happen in the case of invalid `MarkupOrComponentOrHTML` (in which case we should panic), or in the case of `Markup`.

If this is for the case of `Markup` type, then I think that:

```
If(false, Tag("p"), If(true, Tag("p"))) // produces -> [Tag("noscript")]
If(true, Tag("p"), If(true, Tag("p"))) // produces -> [Tag("noscript"), If(true, Tag("p"))]
```

Which surely can't be what we want here?